LayoutTests/fullscreen/full-screen-keyboard-lock-option-enabled.html

```diff
@@ -0,0 +1,57 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <title>This tests that full screen mode with keyboard lock option allows ESC key event.</title>
+</head>
+<body>
+    <div style="container-type: size; width: 100%; height: 100%;" id="target">
+        <p>Test passes if ESC key received by the test case after enter to fullscreen.</p>
+    </div>
+    <script src="full-screen-test.js"></script>
+    <script>
+
+        window.internals.settings.setFullScreenKeyboardLock(true);
+
+        var callback;
+        var fullscreenChanged = function(event)
+        {
+            if (callback)
+                callback(event)
+        };
+
+        var spanEnteredFullScreen = function() {
+            callback = null;
+            testExpected("document.webkitIsFullScreen", true);
+            if (window.testRunner && eventSender.keyDown) {
+                waitForEvent(document, 'keypress', keyPressed);
+                eventSender.keyDown("escape");
+            }
+        };
+
+        var keyPressed = function(event) {
+            testExpected('event.keyCode', 27); // 'Escape'
+            endTest();
+        }
+
+        async function test() {
+            await new Promise((resolve) => {
+                callback = spanEnteredFullScreen;
+
+                waitForEvent(document, 'webkitfullscreenchange', fullscreenChanged);
+                addEventListener("fullscreenchange", resolve, { once: true });
```

Comment:
Kind of strange that you're missing promises and event listeners here, can you only use one of them?

It doesn't seem like the promise is needed here.

Reply:
@nt1m  Thanks for review the codes. Will update the test cases.